app/models/job_application.rb

```diff
@@ -135,18 +135,8 @@ def active_status?
     INACTIVE_STATUSES.exclude?(status)
   end
 
-  Document = Data.define(:filename, :data)
-
-  def submitted_application_form
-    if vacancy.uploaded_form?
-      return Document["no_application_form.txt", "the candidate has no application for on record"] unless application_form.attached?
-
-      extension = File.extname(application_form.filename.to_s)
-      Document["application_form#{extension}", application_form.download]
-    else
-      pdf = JobApplicationPdfGenerator.new(self).generate
-      Document["application_form.pdf", pdf.render]
-    end
+  def has_pre_interview_checks?
+    status.in?(%w[interviewing unsuccessful_interview offered declined])
```

Comment:
We've had a few bugs related to status management when we employ this imperative method definition. Whenever we update `STATUS_TRANSITION`  we risk introducing a bug.

We could instead define a constant that is populated with all the statuses after `shortlisted` from `STATUS_TRANSITION`

```suggestion
 pre_shortlisted, post_shortlisted = JobApplication::STATUS_TRANSITIONS.keys.split { it == "shortlisted" }
POST_SHORTLISTED = post_shortlisted.flat_map { JobApplication::STATUS_TRANSITIONS[it] }.uniq - ["withdrawn"]

status.in?(POST_SHORTLISTED)
```